packages/gatsby/src/query/query-compiler.js

```diff
@@ -93,20 +88,28 @@ class Runner {
     this.parentSpan = parentSpan
   }
 
-  reportError(message) {
-    const queryErrorMessage = `${report.format.red(`GraphQL Error`)} ${message}`
-    if (process.env.gatsby_executing_command === `develop`) {
-      websocketManager.emitError(overlayErrorID, queryErrorMessage)
-      lastRunHadErrors = true
+  async compileAll(activity) {
+    const messages = []
+
+    let nodes = await this.parseEverything(messages)
```

Comment:
We could, but it will need some more refactoring - how about passing `addError` callback (that would just do `messages.push` instead of `messages`, then all the functions in the chain can pass that callback around instead of passing mutable `messages`?

Reply:
i like that better but no need to do this in this PR, i'm happy to get this shipped